File: server/db/models/artwork.py
from sqlalchemy import Integer, String, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from ..connection import db

class Artwork(db.Model):
    id: Mapped[int] = mapped_column(primary_key=True)
    title: Mapped[str] = mapped_column(unique=True)
    price: Mapped[float] = mapped_column(nullable=True)
    medium: Mapped[str] = mapped_column(nullable=True)
    artist: Mapped[int] = mapped_column(ForeignKey("user.id"))

    def as_dict(self):
        return {c.name: getattr(self, c.name) for c in self.__table__.columns}

File: node/models/artwork.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Artwork extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate({User}) {
      // define association here
      Artwork.belongsTo(User, {foreignKey: 'artist'})
    }
  }
  Artwork.init({
    title: DataTypes.STRING,
    price: DataTypes.DECIMAL,
    medium: DataTypes.STRING,
    artist: {
      type: DataTypes.INTEGER,
      references: 'users',
      referencesKey: 'id',
    }
  }, {
    sequelize,
    modelName: 'artwork',
  });
  return Artwork;
};